src/main/java/dk/kb/discover/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,117 @@
+package dk.kb.discover.webservice;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dk.kb.util.Resolver;
+import dk.kb.util.yaml.YAML;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OpenApiResource {
+
+    public static final String APPLICATION_YAML = "application/yaml";
+
+    /**
+     * Pattern to allow search-replace for variabels defined as ${config.yaml.path} in OpenAPI specifications.
+     * Everything after 'config.' is treated as a path to an entry in the backing configuration.
+     */
+    private static final Pattern CONFIG_REPLACEMENT= Pattern.compile("\\$\\{config\\.([^}]+)\\}");
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a YAML file.
+     */
+    @GET
+    @Produces(APPLICATION_YAML)
+    @Path("/ds-discover-openapi_v1.yaml")
+    public Response getYamlSpec() {
+        InputStream yamlStream = Resolver.openFileFromClasspath("ds-discover-openapi_v1.yaml");
+        String openApiSpec = YAML.parse(yamlStream).toString();
```

Comment:
Why is this passed through the YAML-parser first? Is it to remove comments? It is my understanding that we consider the raw OpenAPI YAML as publicly available and as such it would be nice to keep comments and original formatting.

Reply:
My brain is still in YAML-land. You are right, this is not needed. I'll remove it.